### PR TITLE
Allow copy operations to work on multiple selected functions

### PR DESCRIPTION
**Pull request by paulsapps**
Allows "copy name", "copy address" to work with multiple selected functions. Also adds "copy name and address" for single/multiple functions.

### CODE REVIEW

File: plugin/lighthouse/ui/coverage_overview.py [hunk 1]
```diff
@@ -491,32 +497,36 @@ def _process_ctx_menu_action(self, action):
...
-        # handle the 'Copy name' action
-        elif action == self._action_copy_name:
-            name_index = self._model.index(index.row(), FUNC_NAME)
-            function_name = self._model.data(name_index, QtCore.Qt.DisplayRole)
+        # handle the 'Rename' action (only applies to a single function)
+        if action == self._action_rename and len(selected_rows) == 1:
+            gui_rename_function(function_addresses[0])
+
+        # handle the 'Copy name and address' or 'Copy names and addresses' action
+        elif action == self._action_copy_name_and_address or action == self._action_copy_names_and_addresses:
```
Comment (gaasedelen) [inline plugin/lighthouse/ui/coverage_overview.py]: Please change to:

```
elif action in [self._action_copy_name_and_address, self._action_copy_names_and_addresses]:
    ...
```

File: plugin/lighthouse/ui/coverage_overview.py [hunk 2]
```diff
@@ -491,32 +497,36 @@ def _process_ctx_menu_action(self, action):
...
+            for idx, val in enumerate(selected_rows):
+                name_index = self._model.index(val.row(), FUNC_NAME)
+                function_name_and_address += "0x%X" % function_addresses[idx]
+                function_name_and_address += " "
+                function_name_and_address += self._model.data(name_index, QtCore.Qt.DisplayRole)
+                function_name_and_address += "\n"
+            copy_to_clipboard(function_name_and_address)
+
+        # handle the 'Copy name' or 'Copy names' action
+        elif action == self._action_copy_name or action == self._action_copy_names:
```
Comment (gaasedelen) [inline plugin/lighthouse/ui/coverage_overview.py]: Same as comment above,

```
elif action in [self._action_copy_name, self._action_copy_names]:
    ...
```

File: plugin/lighthouse/ui/coverage_overview.py [hunk 3]
```diff
@@ -491,32 +497,36 @@ def _process_ctx_menu_action(self, action):
...
+                name_index = self._model.index(val.row(), FUNC_NAME)
+                function_name += self._model.data(name_index, QtCore.Qt.DisplayRole)
+                function_name += "\n"
             copy_to_clipboard(function_name)
 
-        # handle the 'Copy address' action
-        elif action == self._action_copy_address:
-            address_string = "0x%X" % function_address
+        # handle the 'Copy address' or 'Copy addresses' action
+        elif action == self._action_copy_address or action == self._action_copy_addresses:
```
Comment (gaasedelen) [inline plugin/lighthouse/ui/coverage_overview.py]: Same as the other two comments!

File: plugin/lighthouse/ui/coverage_overview.py [hunk 4]
```diff
@@ -327,9 +327,14 @@ def _ui_init_ctx_menu_actions(self):
 
         # function actions
         self._action_rename = QtWidgets.QAction("Rename", None)
+        self._action_copy_name_and_address = QtWidgets.QAction("Copy name and address", None)
```
Comment (gaasedelen) [inline plugin/lighthouse/ui/coverage_overview.py]: Move this action/member definition to be after `self._action_copy_address`

File: plugin/lighthouse/ui/coverage_overview.py [hunk 5]
```diff
@@ -327,9 +327,14 @@ def _ui_init_ctx_menu_actions(self):
...
         self._action_copy_name = QtWidgets.QAction("Copy name", None)
         self._action_copy_address = QtWidgets.QAction("Copy address", None)
 
+        self._action_copy_names_and_addresses = QtWidgets.QAction("Copy names and addresses", None)
```
Comment (gaasedelen) [inline plugin/lighthouse/ui/coverage_overview.py]: Same as above, move this definition after `self._action_copy_addresses`

**Comment by gaasedelen** [comment]
This seems like a reasonable request to me :-)

I've made a few nitpicks on code style and readability, but nothing major. I'd  also like you to make the PR to the [develop](https://github.com/gaasedelen/lighthouse/tree/develop) branch where I work on vNext / new features. Master is reserved for major releases or hotfixes.

Thanks for the PR!

**Comment by paulsapps** [comment]
Fixed.. I think :)?

**Comment by gaasedelen** [comment]
Thanks, looks good to me 👍

**Pull request merged**